### ISSUE TITLE
standalone converters for hmp and hmi?

### ISSUE THREAD

**Issue by sezero** [Mindwerks/wildmidi#181]
@psi29a:  Do you plan on adding standalone hmp and hmi converters like
those we have for mus and xmi? I vaguely remember that you had intended
to do so quite some time ago (I can't find the relevant thread at all),
but it would be nice if we had them.

**Comment by psi29a** [comment]
I originally wanted just one decoder per format, not two... :/

**Comment by sezero** [comment]
Well, I do trust my memory but I can't find the reference for it :/

Your versions have the advantage that one doesn't have to initialize the
library for them and are very portable across mother projects. That's why
I wanted to have hmp and hmi versions of them.

**Comment by psi29a** [comment]
Me too...

**Comment by sezero** [comment]
Making this a wishlist item.

**Comment by sezero** [comment]
> I'll try to implement one one of these days. ;)
> Sadly work, house and 4 children make it hard to find time.

Had any luck looking into this?

**Comment by psi29a** [comment]
No, still for the wishlist. I still want to do it but anyone else wants to, they are free to do so. ;)

**Comment by psi29a** [comment]
I take that back... i might have something...

**Comment by sezero** [comment]
> I take that back... i might have something...

OK, I'd like to see when they're available.

**Comment by psi29a** [comment]
I know it's been forever... but I'm really tired and this is just baffling me:
```
wildmidi -x ./03\ -\ At\ Doom\'s\ Gate.mus
ERROR: No midi file given
wildmidi -x ./03\ -\ At\ Doom\'s\ Gate.mus test.mid
Conversion failed: System Error (_WM_BufferFile:216) test.mid (Unable to stat) : No such file or directory.
```

Derp... I seem to remember this working and dump directly to file. :/

**Comment by sezero** [comment]
> I know it's been forever... but I'm really tired and this is just baffling me:
> 
> ```
> wildmidi -x ./03\ -\ At\ Doom\'s\ Gate.mus
> ERROR: No midi file given
> wildmidi -x ./03\ -\ At\ Doom\'s\ Gate.mus test.mid
> Conversion failed: System Error (_WM_BufferFile:216) test.mid (Unable to stat) : No such file or directory.
> ```
> 
> Derp... I seem to remember this working and dump directly to file. :/

Move the target output midi name right after -x:
`wildmidi -x test.mid ./03\ -\ At\ Doom\'s\ Gate.mus`

**Comment by psi29a** [comment]
RIght... O.o  I got something to compile with without errors to support HMP but it doesn't 'hooked' up yet to properly count the buffer to give a proper outsize. It was originally meant to write to file a few bytes at a time, so I'm tricking it to write to a buffer.

It's LGPLv2 code from Descent (DX1).  It should give us a baseline for comparison to our in-house f_hmp.c
